IPython/core/completer.py

```diff
@@ -344,11 +344,55 @@ def __eq__(self, other)->Bool:
     def __hash__(self):
         return hash((self.start, self.end, self.text))
 
+
 _IC = Iterator[Completion]
 
-def rectify_completions(text:str, completion:_IC, *, _debug=False)->_IC:
+
+def _deduplicate_completions(text: str, completions: _IC)-> _IC:
     """
-    Rectify a set of completion to all have the same ``start`` and ``end``
+    Deduplicate a set of completions.
+
+    .. warning:: Unstable
+
+        This function is unstable, API may change without warning.
+
+    Parameters
+    ----------
+    text: str
+        text that should be completed.
+    completions: Iterator[Completion]
+        iterator over the completions to deduplicate
+
+
+    Completions coming from multiple source, may be different but end up having
+    the same effect when applied to ``text``. If this is the case, this will
+    consider completions as equal and only emit the first encountered.
+
+    Not folded in `completions()` yet for debug purpose, and to detect when the
+    IPython completer does return things that Jedi does not, but should be at
+    some point.
+    """
+    completions = list(completions)
+    if not completions:
+        return
+
+    starts = (c.start for c in completions)
+    ends = (c.end for c in completions)
+
+    new_start = min(starts)
+    new_end = max(ends)
```

Comment:
Since `starts` and `ends` aren't really needed further, I think it would be clearer/simpler to write

```python
new_start = min(c.start for c in completions)
new_end = max(c.end for c in completions)
```

no?